stored_requests/cache/cacher/cacher.go

```diff
@@ -0,0 +1,74 @@
+package cacher
+
+import (
+	"errors"
+)
+
+// Cacher is the cacher interface for the db_fetchers
+type Cacher interface {
+	// Configure opens a new connection to the cache database.
+	// This should be called once
+	Configure(settings *Settings) error
+
+	// Name of the cache client
+	Name() string
+
+	// Ping cache database
+	Ping() error
+
+	// Close connection to cache database
+	Close()
+
+	// Get a string based off of a key
+	Get(key string) (string, error)
+
+	// Set/Update a value with TTL in seconds
+	Set(key, value string, ttl uint) error
+}
+
+var (
+	// ErrDoesNotExist
+	ErrDoesNotExist = errors.New("cacher: this key does not exist.")
+	// DefaultTTL
+	// TODO: what default TTL should we use?
+	DefaultTTL = uint(60 * 60 * 24)
+)
+
+// Settings is used to configure your cache backend.
+type Settings struct {
```

Comment:
I don't think a single `Settings` struct like this will hold up very well. You can already see some cracks in the design because the `inmemorycache` doesn't use most of them. As new cache implementations get added, this will become quite messy.

On a more broad note, `Configure` functions like this are bug-prone. Code like this:

```
thing := someType{}
thing.Configure(options)
```

can always be simplified to:

```
thing := someType{
  cfg: options
}
```

The latter is significantly safer because you _can't_ "forget" to construct the object. If you do, your code simply won't compile. You _can_, however, forget to call `Configure()`. In this case, it's also better because the args at construction time are type-specific. The `inmemorycache.client` can define _only_ `Size`, while the `rediscache.client` can define everything it needs. As more backends get added, we won't be tied to a global data structure with _all_ the config options.

Reply:
The reason why I did that was so that the code isn’t aware of the specific structures in each of the cache libraries. What I tried to do was take advantage of Go’s blank import when writing the cache drivers  https://golang.org/doc/effective_go.html#init which means using getters/setters such as the Configure method

Reply:
> On a more broad note, Configure functions like this are bug-prone. Code like this:

```
thing := someType{}
thing.Configure(options)
can always be simplified to:

thing := someType{
  cfg: options
}
```

I agree that this is the best way of dealing with structs when they are exported, but the structs inside of each cache driver are unexported and are not accessible by the rest of the PBS code https://github.com/prebid/prebid-server/pull/262/commits/fa883640251577e166fc2a8aa1685dfe90579caf#diff-479f8c90ce1a42e8730dbcb8aa2d004cR8